fog/sample-paykit/proto/remote_wallet.proto

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+/// This file contains a simple service wrapper for a wallet that is used by the fog conformance tests
+/// to perform balance checks against an arbitrary client implementation.
+/// The sample-paykit includes such implementation, and the idea is that other clients/SDKs could
+/// also implement this service and then get tested by the fog conformance tests Python script.
+
+syntax = "proto3";
+
+package remote_wallet;
+
+option java_package = "com.example.myapplication.proto";
+option java_outer_classname = "RemoteWalletProto";
+
+import "google/protobuf/empty.proto";
+
+service RemoteWalletApi {
+    /// Start a new client and perform a fresh balance check.
+    rpc FreshBalanceCheck (FreshBalanceCheckRequest) returns (BalanceCheckResponse);
+
+    /// Perform a balance check on a previously-started client.
+    rpc FollowupBalanceCheck (FollowupBalanceCheckRequest) returns (BalanceCheckResponse);
+
+    /// Stop a previously-started client.
+    rpc Stop (StopRequest) returns (google.protobuf.Empty);
+
+    /// Get debug info for a previously-started client.
+    rpc Debug (DebugRequest) returns (DebugResponse);
+
+    /// Drop all previously-started clients.
+    rpc Reset (google.protobuf.Empty) returns (google.protobuf.Empty);
+}
+
+message FreshBalanceCheckRequest {
+    /// Root entropy for the account the new client will be using.
+    bytes root_entropy = 1;
+
+    /// View server URI
+    string view_server_uri = 2;
+
+    /// Ledger server URI
+    string ledger_server_uri = 3;
+}
+
+message BalanceCheckResponse {
+    /// A unique client identifier that is used in followup requests to refer to the client.
+    uint32 client_id = 1;
+
+    /// The account balance.
+    uint64 balance = 2;
```

Comment:
Not sure if this is entirely critical, but technically an account could have a balance of `> uint64.max`.

A `uint64` + an additional `uint8` would be required to hold the largest possible balance. This is of course probably very unlikely to ever happen in production though, so it might not be a thing we want to worry about. However, for comparison, the Swift SDK does make it a point to handle account balances `> uint64.max` though.

Reply:
Since the idea is to use this with the conformance test script, and the amounts are controlled by it, I think it's safe to not worry about >uint64.max and keep this thing simple.